core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java

```diff
@@ -70,35 +71,69 @@ private FeatureFlaggingRegistry() {}
    * @param feature   Name representing the registered feature
    * @param condition This predicate will be evaluated at deployment time. The {@link MuleContext} corresponds to the context that
    *                  is being created for this application.
+   * @deprecated Use {@link #registerFeatureFlag(Feature, Predicate)} instead.
    */
+  @Deprecated
   public void registerFeature(Feature feature, Predicate<MuleContext> condition) {
+    validate(feature, condition);
+    Predicate<MuleContext> added = configurations.putIfAbsent(feature, condition);
+    if (added != null) {
+      throw new IllegalArgumentException(format(FEATURE_ALREADY_REGISTERED, feature));
+    }
+  }
+
+  /**
+   * Registers a {@link Predicate} associated with a String which represents a given feature. The {@link Predicate} will be
```

Comment:
what do you mean by `associated with a String`? where is a String here?

Reply:
I just copied the deprecated method documentation, but this is a confusing text indeed... will update both.